apps/prairielearn/src/question-servers/calculation.js

```diff
@@ -89,8 +89,18 @@ function questionFunctionExperiment(name, control, candidate) {
         // but not for the control. To avoid false positives, we'll only compare
         // the "data" portion of the results. The "course issues" portion of the
         // results was already handled above.
-        const controlData = controlResult?.[1];
-        const candidateData = candidateResult?.[1];
+        //
+        // We round-trip both data objects through JSON to strip out anything that
+        // couldn't be natively represented in JSON. This is important because
+        // the control implementation will return values from within the same process,
+        // whereas the candidate implementation will always be serialized to and
+        // deserialized from JSON, which will lose information about properties
+        // that can't be represented in JSON.
+        //
+        // Losing this information is fine in practice, as this data is all
+        // ultimately persisted in the database as JSON.
+        const controlData = JSON.parse(JSON.stringify(controlResult?.[1]));
+        const candidateData = JSON.parse(JSON.stringify(candidateResult?.[1]));
```

Comment:
This will avoid more false positives.

I'm wondering if we should do this round-trip lower in the stack - that is, in `calculation-inprocess`. That way everything above it doesn't have to care about this distinction. See also: https://github.com/PrairieLearn/PrairieLearn/pull/8376

Reply:
I'm certainly in favor of moving this as low in the stack as possible. Bonus points if we can somehow use typing to enforce it.

Reply:
Unfortunately there's no type for "this thing was round-tripped through JSON", but yeah, I can move it lower down 🙏 

If you're talking about `zod`, this data come directly from user-authored JavaScript functions, so it won't have a well-known type that we can parse it to.

Reply:
I pushed this lower in the stack in e36c46d.